homeassistant/helpers/label_registry.py

```diff
@@ -188,10 +190,11 @@ def async_update(
         if not changes:
             return old
 
+        self.hass.verify_event_loop_thread("async_update")
```

Comment:
This one is a bit later on purpose since the dict lookup is thread safe and there is no need to incur the cost of the thread check until we are actually going to do something unsafe to do in a thread